src/tests/sqlsmith/src/sql_gen/types.rs

```diff
@@ -0,0 +1,281 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! This module contains datatypes and functions which can be generated by sqlsmith.
+
+use std::collections::HashMap;
+use std::sync::{Arc, LazyLock};
+
+use risingwave_common::types::struct_type::StructType;
+use risingwave_common::types::{DataType as RwDataType, DataTypeName};
+use risingwave_expr::expr::AggKind;
+use risingwave_expr::sig::agg::{agg_func_sigs, AggFuncSig as RwAggFuncSig};
+use risingwave_expr::sig::cast::{cast_sigs, CastContext, CastSig as RwCastSig};
+use risingwave_expr::sig::func::{func_sigs, FuncSign as RwFuncSig};
+use risingwave_frontend::expr::ExprType;
+use risingwave_sqlparser::ast::{DataType as AstDataType, StructField};
+
+pub static STRUCT_FIELD_1: &str = "a";
+
+/// Datatypes which can be generated by sqlsmith
+#[derive(Debug, PartialEq, Hash, Eq, Clone, Copy)]
+pub enum DataType {
```

Comment:
Agree with `since there are infinitely many struct and list expressions which can be generated.`,

I am thinking if we can limit the types of expressions by controlling the depth of the expression,
for example, struct of list, list of struct, and nested struct(but only nested once) are commonly seen, but we forbid more complicated types.

I am not sure if the approach above can make things easier, otherwise, correct me if I am wrong, it seems we will also need to write
`ListOfFloat`, `ListOfStructOfInt` etc.? Consider we add one more primitive type in the future, we will need to add a lot of associated `List` and `Struct` types.

Reply:
Or `add a lot of associated List and Struct types` cannot be avoided anyway?

Reply:
> Agree with `since there are infinitely many struct and list expressions which can be generated.`,
> 
> I am thinking if we can limit the types of expressions by controlling the depth of the expression, for example, struct of list, list of struct, and nested struct(but only nested once) are commonly seen, but we forbid more complicated types.
> 
> I am not sure if the approach above can make things easier, otherwise, correct me if I am wrong, it seems we will also need to write `ListOfFloat`, `ListOfStructOfInt` etc.? Consider we add one more primitive type in the future, we will need to add a lot of associated `List` and `Struct` types.

Hmm I think your suggestion might work better. Let me try it.